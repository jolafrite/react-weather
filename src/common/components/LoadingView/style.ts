import { makeStyles, createStyles } from "@material-ui/core";
import { rem } from "polished";

export const useStyles = makeStyles(
  createStyles({
    root: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      height: '100%',
      width: '100%',
    },
    card: {
      padding: rem(10),
    },
    progress: {
      height: rem(50),
      width: rem(50),
    }
  })
);
