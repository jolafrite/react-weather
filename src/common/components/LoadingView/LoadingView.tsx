import React, { FC, memo } from "react";
import { CircularProgress, Card } from "@material-ui/core";
import Layout from "../Layout";
import { useStyles } from "./style";

export interface ILoadingViewProps {}

const LoadingView: FC<ILoadingViewProps> = () => {
  const classes = useStyles();

  return (
    <Layout>
      <div className={classes.root}>
        <Card className={classes.card}>
          <CircularProgress
            size=""
            className={classes.progress}
          />
        </Card>
      </div>
    </Layout>
  )
};

export default memo(LoadingView);
