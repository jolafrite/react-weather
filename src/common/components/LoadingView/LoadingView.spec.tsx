import React from 'react';
import { cleanup, render } from "@testing-library/react";
import LoadingView, { ILoadingViewProps } from "./LoadingView";

jest.mock('../Layout', () => ({ children }: any) => <div className="Layout">{children}</div>);

describe('component: LoadingView', () => {

  const createComponent = (values = {}) => {
    const props: ILoadingViewProps = {
      ...values,
    };

    return render(
      <LoadingView
        {...props}
      />
    );
  };

  afterEach(cleanup);

  test('default render', () => {
    const component = createComponent();

    expect(
      component.container
    ).toMatchSnapshot();
  });

});
