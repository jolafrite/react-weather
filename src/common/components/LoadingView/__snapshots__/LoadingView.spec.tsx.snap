// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: LoadingView default render 1`] = `
<div>
  <div
    class="Layout"
  >
    <div
      class="makeStyles-root-1"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-2 MuiPaper-rounded"
      >
        <div
          class="MuiCircularProgress-root makeStyles-progress-3 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
